hawkular-inventory-rest-api/src/main/java/org/hawkular/inventory/rest/RestResourcesMetrics.java

```diff
@@ -64,17 +67,26 @@
 
     @POST
     @javax.ws.rs.Path("/{environmentId}/resources/{resourcePath:.+}/metrics")
-    @ApiOperation("Associates a pre-existing metric with a resource")
+    @ApiOperation("Either creates a new metric owned by the resource or associates a pre-existing metric with the " +
+            "resource. This depends on what you pass as the the body of the request. A JSON array of strings is " +
+            "understood as a list of pre-existing metric paths that are associated with the resource. If the body is" +
+            " a JSON object or an array of JSON objects, the new metric or metrics are created \"underneath\" the " +
+            "resource.")
     @ApiResponses({
-            @ApiResponse(code = 204, message = "OK"),
-            @ApiResponse(code = 404, message = "Tenant, environment, resource or metric doesn't exist",
+            @ApiResponse(code = 201, message = "New metric created under the resource"),
+            @ApiResponse(code = 204, message = "Existing metric successfully associated"),
+            @ApiResponse(code = 404, message = "Tenant, environment, resource doesn't exist. Also when an array of " +
+                    "strings is supplied and one of the metrics in that array doesn't exist.",
                     response = ApiError.class),
             @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
     })
-    public Response associateMetrics(@PathParam("environmentId") String environmentId,
-            @Encoded @PathParam("resourcePath") String resourcePath,
-            @ApiParam("A list of paths to metrics to be associated with the resource. They can either be canonical or" +
-                    " relative to the resource.") Collection<String> metricPaths) {
+    public Response associateOrCreateMetrics(@PathParam("environmentId") String environmentId,
+                                             @Encoded @PathParam("resourcePath") String resourcePath,
+                                             @ApiParam("This is either a metric blueprint or a list of paths to " +
+                                                     "metrics to be associated with the resource. They can either be " +
+                                                     "canonical or relative to the resource.")
+                                                 Object metricPathsOrBlueprint,
+                                             @Context UriInfo uriInfo) {
```

Comment:
No. If you look at the new `createOrAssociateMetric()`, you'll see that it basically is a switch that 1) handles a list of metric paths as used to be the case and in addition it handles adding a blueprint. I.e. you can send 2 types of data to this one endpoint, either `["path","path"]` or `{"id": "id"}`.